### PR TITLE
Check that value produced by byteArrayToBigInt does not exceed dlog group order

### PR DESCRIPTION
**Pull request by greenhat**
Ref https://github.com/ScorexFoundation/sigmastate-interpreter/issues/311

### CODE REVIEW

File: sigma-impl/src/test/scala/special/sigma/BasicOpsTests.scala [hunk 1]
```diff
@@ -18,6 +18,11 @@ class BasicOpsTests extends FunSuite with ContractsTestkit with Matchers {
     SigmaDsl.byteArrayToBigInt(collection[Byte](-1)).signum shouldBe 1
   }
 
+  test("ByteArrayToBigInt should always produce big int less than dlog group order") {
+    an [RuntimeException] should be thrownBy
+      SigmaDsl.byteArrayToBigInt(Cols.fromArray(Array.fill[Byte](500)(1)))
```
Comment (aslesarenko) [inline sigma-impl/src/test/scala/special/sigma/BasicOpsTests.scala]: @greenhat I suggest to add tests which also check behavior on boundary cases 
dlogGroupOrder-1, dlogGroupOrder, dlogGroupOrder+1
